java/org/apache/catalina/core/ApplicationPushBuilder.java

```diff
@@ -26,6 +26,7 @@
 import java.util.Locale;
 import java.util.Map;
 import java.util.Set;
+import java.util.StringJoiner;
```

Comment:
Just a note: this cannot be back-ported to Tomcat 8.5.x because it must be buildable and runnable on Java 7.

Reply:
Thanks, Christopher.